### PR TITLE
Improve docstrings of pyplot axis-related functions

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

This PR improves th docstrings of pyplot axis-related functions:
- `x/yscale`
- `x/ylim`
- `x/ylabel`
- `x/yticks`

### CODE REVIEW

**Comment by timhoffm** [comment]
The failing test is something about latex. Since I've only changed docstring, it must be unrelated to the PR.

**Comment by jklymak** [comment]
Yes if you rebase it’ll pick up the latex fix.   But this could pass w/o it.

**Comment by timhoffm** [comment]
rebased.

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -1632,24 +1652,60 @@ def yscale(*args, **kwargs):
...
 
-    The keyword args, if any, are :class:`~matplotlib.text.Text`
-    properties. For example, to rotate long labels::
+    labels : array_like, optional
+        A list of explicit labels to place at the given *locs*. You can pass
+        an empty list to disable ticks and labels.
 
-      xticks( arange(12), calendar.month_name[1:13], rotation=17 )
+    **kwargs
+        :class:`.Text` properties can be used to control the appearence of
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: appearance

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -1672,24 +1728,60 @@ def xticks(*args, **kwargs):
...
+    ----------
+    locs : array_like
+        A list of positions at which ticks should be placed.
+
+    labels : array_like, optional
+        A list of explicit labels to place at the given *locs*. You can pass
+        an empty list to disable ticks and labels.
+
+    **kwargs
+        :class:`.Text` properties can be used to control the appearence of
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: appearance

File: lib/matplotlib/pyplot.py [hunk 3]
```diff
@@ -1632,24 +1652,60 @@ def yscale(*args, **kwargs):
...
+
+    Set label locations:
+
+        >>> xticks(np.arange(0, 1, step=0.2))
+
+    Set text labels:
+
+        >>> xticks(np.arange(5), ('Tom', 'Dick', 'Harry', 'Sally', 'Sue'))
+
+
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: Add heading `Set text labels and properties:`?

File: lib/matplotlib/pyplot.py [hunk 4]
```diff
@@ -1632,24 +1652,60 @@ def yscale(*args, **kwargs):
...
+    Parameters
+    ----------
+    locs : array_like
+        A list of positions at which ticks should be placed.
 
-    The keyword args, if any, are :class:`~matplotlib.text.Text`
-    properties. For example, to rotate long labels::
+    labels : array_like, optional
+        A list of explicit labels to place at the given *locs*. You can pass
+        an empty list to disable ticks and labels.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: Is the empty list for `labels` or `locs`? The example below passes it as the first parameter (`locs`).

File: lib/matplotlib/pyplot.py [hunk 5]
```diff
@@ -1672,24 +1728,60 @@ def xticks(*args, **kwargs):
...
-    ::
+    Set text labels:
 
-      # return locs, labels where locs is an array of tick locations and
-      # labels is an array of tick labels.
-      locs, labels = yticks()
+        >>> yticks(np.arange(5), ('Tom', 'Dick', 'Harry', 'Sally', 'Sue'))
 
-      # set the locations of the yticks
-      yticks( arange(6) )
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: Add heading `Set text labels and properties:`?

File: lib/matplotlib/pyplot.py [hunk 6]
```diff
@@ -1632,24 +1652,61 @@ def yscale(*args, **kwargs):
...
+    Parameters
+    ----------
+    locs : array_like
+        A list of positions at which ticks should be placed. You can pass an
+        empty list to disable xticks.
 
-    The keyword args, if any, are :class:`~matplotlib.text.Text`
-    properties. For example, to rotate long labels::
+    labels : array_like, optional
+        A list of explicit labels to place the the given *locs*.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: the the -> at the

File: lib/matplotlib/pyplot.py [hunk 7]
```diff
@@ -1672,24 +1729,61 @@ def xticks(*args, **kwargs):
...
+        yticks(locs, [labels], **kwargs)  # Set locations and labels
+
+    Parameters
+    ----------
+    locs : array_like
+        A list of positions at which ticks should be placed. You can pass an
+        empty list to disable yticks.
+
+    labels : array_like, optional
+        A list of explicit labels to place the the given *locs*.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: the the -> at the

**Comment by WeatherGod** [comment]
You should also updated the corresponding `z` functions in mplot3d so that
they match up.

**Comment by timhoffm** [comment]
@WeatherGod updating the `z` functions is not trivial for a bunch of reasons, because the have some differences. I would rather tackle these in a separate pull request.

File: lib/matplotlib/pyplot.py [hunk 8]
```diff
@@ -1500,65 +1500,59 @@ def axis(*v, **kwargs):
...
-          'horizontalalignment' : 'center'
-          }
+    Call signature::
 
-    .. seealso::
+        xlabel(label, fontdict=None, labelpad=None, **kwargs)
 
-        :func:`~matplotlib.pyplot.text`
-            For information on how override and the optional args work
+    This is just a shortcut for calling `.set_xlabel` on the current axes.
```
Comment (story645) [inline lib/matplotlib/pyplot.py]: I feel like this implies that either `ax.xlabel` or `plt.set_xlabel valid`.
Alternative suggestion is something like: this is the pyplot interface equivalent of calling :method:`~matplotlib.axes.Axes.set_xlabel`on the current axes.

File: lib/matplotlib/pyplot.py [hunk 9]
```diff
@@ -1500,65 +1500,59 @@ def axis(*v, **kwargs):
...
+    Call signature::
 
-    .. seealso::
+        ylabel(label, fontdict=None, labelpad=None, **kwargs)
 
-        :func:`~matplotlib.pyplot.text`
-            For information on how override and the optional args
-            work.
+    This is just a shortcut for calling `.set_ylabel` on the current axes.
+    See there for a full parameter description.
```
Comment (story645) [inline lib/matplotlib/pyplot.py]: same as above

File: lib/matplotlib/pyplot.py [hunk 10]
```diff
@@ -1500,65 +1500,59 @@ def axis(*v, **kwargs):
...
-      xmin, xmax = xlim()   # return the current xlim
-      xlim( (xmin, xmax) )  # set the xlim to xmin, xmax
-      xlim( xmin, xmax )    # set the xlim to xmin, xmax
+        xmin, xmax = xlim()  # return the current xlim
+        xlim((xmin, xmax))   # set the xlim to xmin, xmax
+        xlim(xmin, xmax)     # set the xlim to xmin, xmax
 
-    If you do not specify args, you can pass the xmin and xmax as
-    kwargs, e.g.::
+    If you do not specify args, you can alternatively pass *xmin* or *xmax* as
```
Comment (story645) [inline lib/matplotlib/pyplot.py]: I think the alternatively is clearly implied here

File: lib/matplotlib/pyplot.py [hunk 11]
```diff
@@ -1500,65 +1500,59 @@ def axis(*v, **kwargs):
...
 
-    The new axis limits are returned as a length 2 tuple.
+    Returns
+    -------
+    xmin, xmax
+        A tuple of the new x-axis limits.
 
+    Notes
+    -----
+    This is just a shortcut for calling `~.Axes.get_xlim` or `~.Axes.set_xlim`
```
Comment (story645) [inline lib/matplotlib/pyplot.py]: Same as set_xlabel/set_ylable comment. Calling this function with no arguments (e.g. `xlim()`) is the pyplot equivalent of calling `~.Axes.get_xlim` on the current axis. Calling this function with arguments (e.g. xlim(xmin=3, xmax=5) is the pyplot equivalent of calling `~.Axes.set_xlim` on the current axis.

**Comment by tacaswell** [comment]
The mac failure is due to homebrew changing how python3 installs

**Pull request merged**